responses/test_registries.py

```diff
@@ -68,3 +71,91 @@ class CustomRegistry(registries.FirstMatchRegistry):
 
     run()
     assert_reset()
+
+
+class TestOrderedRegistry:
+    def test_invocation_index(self):
+        @responses.activate(registry=OrderedRegistry)
+        def run():
+            responses.add(
+                responses.GET,
+                "http://twitter.com/api/1/foobar",
+                json={"msg": "not found"},
+                status=404,
+            )
+            responses.add(
+                responses.GET,
+                "http://twitter.com/api/1/foobar",
+                json={"msg": "OK"},
+                status=200,
+            )
+            responses.add(
+                responses.GET,
+                "http://twitter.com/api/1/foobar",
+                json={"msg": "OK"},
+                status=200,
+            )
+            responses.add(
+                responses.GET,
+                "http://twitter.com/api/1/foobar",
+                json={"msg": "not found"},
+                status=404,
+            )
+
+            resp = requests.get("http://twitter.com/api/1/foobar")
+            assert resp.status_code == 404
+            resp = requests.get("http://twitter.com/api/1/foobar")
+            assert resp.status_code == 200
+            resp = requests.get("http://twitter.com/api/1/foobar")
+            assert resp.status_code == 200
```

Comment:
Should we also check the response bodies or use more unique status codes to ensure the ordering?

Reply:
Agree, let me set unique codes and bodies for each request-response pair

Reply:
@markstory 
done